docs/source/trainer_cli.rst

```diff
@@ -0,0 +1,156 @@
+Trainer CLI and config files
+----------------------------
+
+Another source of boilerplate code that Lightning can help to reduce is in the
+implementation of training command line tools. Furthermore, it provides a
+standardized way to configure trainings using a single file that includes
+settings for :class:`~pytorch_lightning.trainer.trainer.Trainer` and user
+extended :class:`~pytorch_lightning.core.lightning.LightningModule` and
+:class:`~pytorch_lightning.core.datamodule.LightningDataModule` classes. The
+full configuration is automatically saved in the log directory. This has the
+benefit of greatly simplifying the reproducibility of experiments.
+
+The main requirement for user extended classes to be made configurable is that
+all relevant init arguments must have type hints. This is not a very demanding
+requirement since anyway it is good practice to do anyway. As a bonus if the
+arguments are also described in the docstrings, then the help of the training
+tool will display them.
+
+----------
+
+
+training_cli
+^^^^^^^^^^^^
+
+The case in which the user's :class:`LightningModule` class implements all
+required :code:`*_dataloader` methods, a :code:`trainer.py` tool can be as
+simple as:
+
+.. code-block:: python
```

Comment:
this code with one of PL included model shall run fine, right? consider using `.. testcode::`

Reply:
Yes I was considering using `.. testcode::` but preferred to open the pull request to get first some initial feedback. Will do.

Reply:
Converted the code snippets to `.. testcode::`. It is the first time I do this, so please let me know if I did something wrong.

Reply:
I finally got all `testcode` working.